solidity/contracts/token/HypNative.sol

```diff
@@ -45,39 +46,34 @@ contract HypNative is FungibleTokenRouter, MovableCollateralRouter {
         _MovableCollateralRouter_initialize(_owner);
     }
 
-    /**
-     * @inheritdoc TokenRouter
-     * @dev uses (`msg.value` - `_amount`) as hook payment and `msg.sender` as refund address.
-     */
-    function transferRemote(
+    function _quoteTransferRemote(
         uint32 _destination,
         bytes32 _recipient,
         uint256 _amount
-    ) external payable virtual override returns (bytes32 messageId) {
-        require(msg.value >= _amount, "Native: amount exceeds msg.value");
-        uint256 _hookPayment = msg.value - _amount;
-        return _transferRemote(_destination, _recipient, _amount, _hookPayment);
+    ) internal view override returns (Quote[] memory quotes) {
+        quotes = super._quoteTransferRemote(_destination, _recipient, _amount);
+        assert(quotes[0].token == address(0));
```

Comment:
This check should be moved to the appropriate forge test.

Reply:
Ditto, I don't think you can be asserting this like the above comment